zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/S3NotebookRepo.java

```diff
@@ -89,12 +92,14 @@
   private final String user;
   private final boolean useServerSideEncryption;
   private final ZeppelinConfiguration conf;
+  private boolean saveAndCommit;
 
   public S3NotebookRepo(ZeppelinConfiguration conf) throws IOException {
     this.conf = conf;
     bucketName = conf.getBucketName();
     user = conf.getUser();
     useServerSideEncryption = conf.isS3ServerSideEncryption();
+    saveAndCommit = conf.isPersistOnCommit();
```

Comment:
Good point, actually originally it was implemented in base class as a global parameter, but after the comments and actual goal of this PR of optimizing "saves" for remote storages, it was changed to target specific storages (currently s3 and zhub). Thus if this setting is fine grained per each storage (e.g. can be disabled in s3 and enabled in zhub storages, and absent in all other storages) they need to handle it separately. The original environment variable just gives initial default values that can be changed after start of server (although can't be persistent when stopped). Does it make sense?